src/core/mempool.rs

```diff
@@ -1079,9 +1087,15 @@ impl MemPoolDB {
                            "size" => consider.tx.metadata.len);
                     total_considered += 1;
 
-                    if !todo(clarity_tx, &consider, self.cost_estimator.as_mut())? {
-                        debug!("Mempool iteration early exit from iterator");
-                        break;
+                    // Run `todo` on the transaction.
+                    match todo(clarity_tx, &consider, self.cost_estimator.as_mut())? {
+                        Some(tx_event) => {
```

Comment:
In this case `else if block_limit_hit
                                            == BlockLimitFunction::CONTRACT_LIMIT_HIT` in `build_anchored_block`, you return Some(event) when before Ok(false) was returned. 

This will result in different behavior

Reply:
updated